packages/workflow-core/src/lib/utils/definition-validator/extensions-validator.ts

```diff
@@ -0,0 +1,124 @@
+} from "../../plugins/external-plugin/types";
+import {WorkflowExtensions} from "../../types";
+import {ruleValidator} from "./rule-validator";
+import {isErrorWithMessage} from "@ballerine/common";
+
+export const extensionsValidator = (extensions: WorkflowExtensions, states: StateMachine<any, any, any>['states']) => {
+  extensions.apiPlugins?.forEach(plugin => {
+
+    const pluginKind = (plugin as ISerializableHttpPluginParams).pluginKind;
+    if (
+      pluginKind === 'api' ||
+      pluginKind === 'kyb' ||
+      pluginKind === 'kyc-session' ||
+      pluginKind === 'email' ||
+      pluginKind === 'kyc'
+    ) {
+      validateApiPlugin(plugin as unknown as ISerializableHttpPluginParams, states)
+    }
+
+    if (pluginKind === 'webhook') {
+      validateWebhookPlugin(plugin as unknown as SerializableWebhookPluginParams)
+    }
+  });
+
+  extensions.commonPlugins?.forEach(plugin => {
+    const pluginKind = (plugin as unknown as ISerializableChildPluginParams).pluginKind;
+    if (pluginKind === 'iterative') {
+      validateIterativePlugin(plugin as unknown as SerializableIterativePluginParams, states)
+    }
+  });
+
+  extensions.childWorkflowPlugins?.forEach(plugin => {
+    const pluginKind = (plugin as unknown as ISerializableChildPluginParams).pluginKind;
+    if (pluginKind === 'child') {
+      validateChildPlugin(plugin as unknown as ISerializableChildPluginParams)
+    }
+  });
+}
+const validateApiPlugin = (
+  plugin: ISerializableHttpPluginParams,
+  states: StateMachine<any, any, any>['states']
+): void => {
+  validateTransformers(plugin.name, plugin.request.transform)
+  plugin.response.transform && validateTransformers(plugin.name, plugin.response.transform)
+
+  validatePluginStateAction(plugin.stateNames, states, plugin.successAction, plugin.errorAction)
+};
+const validateWebhookPlugin = (
+  plugin: SerializableWebhookPluginParams,
+): void => {
+  validateTransformers(plugin.name, plugin.request.transform)
+};
+const validateIterativePlugin = (
+  plugin: SerializableIterativePluginParams,
+  states: StateMachine<any, any, any>['states']
+): void => {
+  validateMapping('jmespath', plugin.iterateOn)
+  validatePluginStateAction(plugin.stateNames, states, plugin.successAction, plugin.errorAction)
+};
+const validateChildPlugin = (
+  plugin: ISerializableChildPluginParams
+): void => {
+  plugin.transformers?.forEach(transform => {
+    validateMapping(transform.transformer as Parameters<typeof validateMapping>[0], transform.mapping as Parameters<typeof validateMapping>[1])
+  })
+};
+
+const validateTransformers = (pluginName: string, transformers: SerializableValidatableTransformer['transform']) => {
+  transformers.forEach(transform => {
+    try {
+      transform.transformer === 'jmespath' &&
+      validateMapping('jmespath', transform.mapping as string);
+      transform.transformer === 'json-logic' &&
+      validateMapping('json-logic', transform.mapping as unknown as Record<string, unknown>)
```

Comment:
This block is a bit hard to read. Consider adding spacing and using `if` statements.